design/userdefined-rootdefinehints.md

```diff
@@ -0,0 +1,148 @@
+      * [Status](#status)
+      * [Table of Contents](#table-of-contents)
+      * [Summary](#summary)
+      * [Motivation](#motivation)
+         * [Goals](#goals)
+         * [Non-Goals](#non-goals)
+      * [Proposal](#proposal)
+         * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+         * [Risks and Mitigations](#risks-and-mitigations)
+      * [Design Details](#design-details)
+         * [Work Items](#work-items)
+         * [Dependencies](#dependencies)
+         * [Test Plan](#test-plan)
+         * [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+         * [Version Skew Strategy](#version-skew-strategy)
+      * [References](#references)
+
+<!-- Added by: mikkosest -->
+
+<!--te-->
+
+## Summary
+
+This document explains how rootdevicehints could be given as user defined parameters as part BaremetalHostSpec. The idea is that user would have a possibilty to defines selective constrains for root device selection. Sometimes selective constrains will not be completely deterministic, but it is upto user, which hints to use. 
+In legacy, only way to how user can select root device hints used, is to choose the name of the hardware profile which has hardcoded root device hint values. When exposing root device hints as part of the BMH spec, this will be part of deprication steps for hardware profile.
+
+## Motivation
+
+We need to be able select root device, when there is multiple devices available so that device is not chosen randomly
+
+### Goals
+
+1. Decide root device hints are needed be supported.
+2. Agree on the way haw information is propagated to ironic through BMO
+
+
+### Non-Goals
+
+The complete set of root device hints as input is not needed only those seen important.
+
+## Proposal
+
+### Implementation Details/Notes/Constraints
+
+- BaremetalHostSpec need to be updated to support list of following root device hints:
+* devicename
+* hctl
+* vendor
+* size
+* wwn
+
+Deteriministic selectors. i.e if used together with each other, those should point the same device instance and should not conflict with group selectors below.
+
+Deterministic selectors
+* devicename
+* hctl
+* wwn
+
+Group selectors
+* vendor
+* size
+
+
+The above parameters are optional in BaremetalHostSpec. If user is using legacy Hardware profiles, the root device hints hardcoded to code will be overwritten or appended with proposed root device hints.If there is deterministic hint like devicename or wwn used, then those will dominant selectors over other hints which would be ignored. 
+
+The new type struct added to spec could look like below. In the legacy there is Storage struct used in inventory that has pretty much the same content, but by having a separete struct we can control easily if we want to add or depricate support for some hints. Also size is given in Gigabytes, not like bytes in Storage struct in order to avoids scale typos in size input.
```

Comment:
the new struct type .. In the original inventory code there is an Storage struct ... 
depricate/deprecate
size is given in Gigabytes, while in Storage it is given in bytes

Reply:
Fixed